beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go

```diff
@@ -443,27 +444,35 @@ func (vs *Server) broadcastAndReceiveBlobs(ctx context.Context, sidecars []*ethp
 
 // broadcastAndReceiveDataColumns handles the broadcasting and reception of data columns sidecars.
 func (vs *Server) broadcastAndReceiveDataColumns(ctx context.Context, sidecars []*ethpb.DataColumnSidecar, root [fieldparams.RootLength]byte) error {
-	for i, sidecar := range sidecars {
-		if err := vs.P2P.BroadcastDataColumn(ctx, uint64(i), sidecar); err != nil {
-			return errors.Wrap(err, "broadcast data column")
-		}
+	eg, _ := errgroup.WithContext(ctx)
+	for i, sd := range sidecars {
+		// Copy the iteration instance to a local variable to give each go-routine its own copy to play with.
+		// See https://golang.org/doc/faq#closures_and_goroutines for more details.
+		colIdx := i
+		sidecar := sd
+		eg.Go(func() error {
```

Comment:
Do we still need the to run `s.internalBroadcastDataColumn` in a goroutine here?

```go
func (s *Service) BroadcastDataColumn(ctx context.Context, columnSubnet uint64, dataColumnSidecar *ethpb.DataColumnSidecar) error {
	// Add tracing to the function.
	ctx, span := trace.StartSpan(ctx, "p2p.BroadcastBlob")
	defer span.End()

	// Ensure the data column sidecar is not nil.
	if dataColumnSidecar == nil {
		return errors.Errorf("attempted to broadcast nil data column sidecar at subnet %d", columnSubnet)
	}

	// Retrieve the current fork digest.
	forkDigest, err := s.currentForkDigest()
	if err != nil {
		err := errors.Wrap(err, "current fork digest")
		tracing.AnnotateError(span, err)
		return err
	}

	// Non-blocking broadcast, with attempts to discover a column subnet peer if none available.
	go s.internalBroadcastDataColumn(ctx, columnSubnet, dataColumnSidecar, forkDigest)

	return nil
}
```

Reply:
(We now have a goroutine calling an other goroutine.)